### PR TITLE
branching factor grammar search heuristic

### PR DESCRIPTION
**Pull request by tomsilver**
this is a bad heuristic, but we may want to use it as a baseline in the future.

### CODE REVIEW

File: tests/approaches/test_grammar_search_invention_approach.py [hunk 1]
```diff
@@ -422,3 +426,53 @@ def test_hadd_lookahead_heuristic():
...
+    }
+    dataset = create_dataset(env, next(env.train_tasks_generator()))
+    atom_dataset = utils.create_ground_atom_dataset(dataset,
+        env.goal_predicates | set(candidates))
+    heuristic = _BranchingFactorHeuristic(env.goal_predicates, atom_dataset,
+                                          candidates)
+    holding_h = heuristic.evaluate({Holding})
+    forall_not_covers_h = heuristic.evaluate({forall_not_covers0,
+                                              forall_not_covers1})
+    # This is just to illustrate that the cost for these two bad predicates are
```
Comment (ronuchit) [inline tests/approaches/test_grammar_search_invention_approach.py:475]: costs* ?

**Review by ronuchit — APPROVED** [review]
lgtm thanks

File: tests/approaches/test_grammar_search_invention_approach.py (hunk 1 above)
Comment (tomsilver) [reply tests/approaches/test_grammar_search_invention_approach.py:475]: no I meant the heuristic value for the set containing both heuristics

Comment (tomsilver) [reply tests/approaches/test_grammar_search_invention_approach.py:475]: oh i see the grammar issue, will fix

**Pull request merged**